### ISSUE TITLE
Training GPU hours

### ISSUE THREAD

**Issue by Kairui-SHI** [Reagan1311/Mask2IV#5]
Thanks for your great work. I wonder how many GPU hours it costs to fine-tune the model on DynamiCrafter. I noticed it's on 2 A100 GPUs. Thanks a lot.

**Comment by dovis2000hit** [comment]
I also have the same question. Could you please tell me how many steps of training were conducted to achieve the accuracy reported in the paper?

**Comment by Reagan1311** [comment]
Hi all, on the HOI4D dataset (11,851 training samples), Stage 1 took about 4 days and Stage 2 took around 3 days with batch size 8.

**Comment by fayewong666999** [comment]
> Hi all, on the HOI4D dataset (11,851 training samples), Stage 1 took about 4 days and Stage 2 took around 3 days with batch size 8.
In lvdm/models/ddpm3d_maskgen.py
null_prompt = self.get_learned_conditioning([""])
-         # prompt_imb = torch.where(prompt_mask, null_prompt, cond_emb.detach())
-         prompt_imb = self.get_learned_conditioning([""] * x.size(0)) 
**Is this a bug? The text prompts seem to be replaced entirely with empty characters.**

**Comment by Reagan1311** [comment]
> > Hi all, on the HOI4D dataset (11,851 training samples), Stage 1 took about 4 days and Stage 2 took around 3 days with batch size 8.
> > In lvdm/models/ddpm3d_maskgen.py
> > null_prompt = self.get_learned_conditioning([""])
> 
> * ```
>       # prompt_imb = torch.where(prompt_mask, null_prompt, cond_emb.detach())
>   ```
> * ```
>       prompt_imb = self.get_learned_conditioning([""] * x.size(0)) 
>   ```
> 
> **Is this a bug? The text prompts seem to be replaced entirely with empty characters.**

This appears to be an oversight on my side. I changed the conditioning to empty prompts when training BDv2, since the first stage training of BDv2 uses position-based conditioning and does not require text input.

However, HOI4D first-stage training is text-conditioned and should use the original text prompts. Therefore, this behavior is incorrect for HOI4D and the text conditioning should be restored.

For the second stage, text prompts are intentionally disabled. The video generation model relies only on the mask trajectories as conditioning, so empty prompts are expected there.

I've updated the code accrodingly.

**Comment by fayewong666999** [comment]
Thank you very much for your prompt reply. Here is my understanding: (1) For HOI4D, the first stage is text-conditioned trajectory generation, so text is required; (2) For BDv2, the first stage is position-conditioned trajectory generation (i.e., the first and last frames), so no text is needed; (3) For the video generation in the second stage, the control conditions come from the trajectory output of the first stage, so no text is required in all cases. Is my understanding correct?

**Comment by Reagan1311** [comment]
> Thank you very much for your prompt reply. Here is my understanding: (1) For HOI4D, the first stage is text-conditioned trajectory generation, so text is required; (2) For BDv2, the first stage is position-conditioned trajectory generation (i.e., the first and last frames), so no text is needed; (3) For the video generation in the second stage, the control conditions come from the trajectory output of the first stage, so no text is required in all cases. Is my understanding correct?

Yes, your understanding is completely correct.

**Comment by fayewong666999** [comment]
**Please forgive my asking again. For the segmentation mask of BDv2, are you using func/bdv2_groundedsam2.py? It seems that this method may result in many masks with holes. Do you have any further post-processing methods?**

**Comment by Reagan1311** [comment]
Yes, we use Grounded-SAM to produce masks from BDv2 dataset. The masks can sometimes be noisy and may contain holes when motion is fast or there are heavy occlusions.

We don’t apply additional post-processing. You could try SAM3 or simple post-processing to improve results.

**Comment by fayewong666999** [comment]
**Dear Authors, after using the official weights you provided to generate videos on the BDv2 dataset, we obtained an FVD of 183 (while your paper reported 155.73). We would like to ask you a few questions: (1) The original resolution of BDv2 is 480x640, but after resizing, the shorter side becomes only 426, which is less than 320x512. (1) How do you handle resolution? Do you simply fill the missing areas with black?  
(2) Would it be possible for you to provide the pseudo-masks of the BDv2 dataset after being processed by Grounded-SAM?  
(3) Why is there a *1.2 scaling operation applied to the HOI4D dataset, but not to the BDv2 dataset?  
(4) Is the FVD calculated based on a resolution of 320x512, meaning both real and synthesized videos are resized to 320x512? We would greatly appreciate it if you could share the detailed evaluation code.**

**Comment by Reagan1311** [comment]
(1) Yes. For BDv2, we pad the missing regions with black pixels, so black borders may appear on the left and right.

(2) Unfortunately, I no longer have access to the previous server, so I am unable to provide the processed pseudo-masks.

(3) The 1.2 scaling is only used for HOI4D because its views are much wider; zooming in helps emphasize the hands and interacted objects.

(4) Yes. FVD is computed at 320×512, with both real and generated videos resized/padded to this resolution.
